### PR TITLE
List PickleProtocol (PCK)

### PR DESCRIPTION
**Pull request by lilcryptomoon**

### CODE REVIEW

**Comment by lilcryptomoon** [comment]
List PickleProtocol [PCK]
- PickleProject.org
- http://31.220.61.164:3001

**Comment by lilcryptomoon** [comment]
List PickleProtocol [PCK]
- PickleProject.org
- http://31.220.61.164:3001

File: assets/src/main/java/bisq/asset/coins/PickleProtocol.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+package bisq.asset.coins;
+
+import bisq.asset.Coin;
+import bisq.asset.DefaultAddressValidator;
+
+public class PickleProtocol extends Coin {
+    public PickleProtocol() {
+
+        super("PickleProtocol", "PCK", new DefaultAddressValidator());
```
Comment (blabno) [inline assets/src/main/java/bisq/asset/coins/PickleProtocol.java]: Do not use deprecated validator

**Review by blabno — CHANGES_REQUESTED** [review]
NACK per comments

File: assets/src/main/java/bisq/asset/coins/PickleProtocol.java (hunk 1 above)
Comment (lilcryptomoon) [reply assets/src/main/java/bisq/asset/coins/PickleProtocol.java]: should I be  using ?
package bisq.asset.coins;

import bisq.asset.Base58BitcoinAddressValidator;
import bisq.asset.Coin;
import bisq.asset.NetworkParametersAdapter;

public class PickleProtocol extends Coin {

    public PickleProtocol() {
        super("PickleProtocol", "PCK", new Base58BitcoinAddressValidator(new PickleProtocolParams()));
    }


    public static class PickleProtocolParams extends NetworkParametersAdapter {

        public PickleProtocolParams() {
            addressHeader = 54;
            p2shHeader = 119;
            acceptableAddressCodes = new int[]{addressHeader, p2shHeader};
        }
    }
}

File: assets/src/main/resources/META-INF/services/bisq.asset.Asset [hunk 2]
```diff
@@ -40,6 +40,7 @@ bisq.asset.coins.MonetaryUnit
 bisq.asset.coins.MoX
 bisq.asset.coins.Namecoin
 bisq.asset.coins.Neos
+bisq.asset.coins.Noir
```
Comment (blabno) [inline assets/src/main/resources/META-INF/services/bisq.asset.Asset:46]: Are you listing more than one coin?

**Review by blabno — CHANGES_REQUESTED** [review]
NACK per comments

**Comment by blabno** [comment]
Also keep the entire PR as a single commit.

**Comment by lilcryptomoon** [comment]
No I am just listing Pickle Protocol

File: assets/src/main/resources/META-INF/services/bisq.asset.Asset (hunk 2 above)
Comment (blabno) [reply assets/src/main/resources/META-INF/services/bisq.asset.Asset:46]: > No I am just listing Pickle Protocol

So this line should not appear in diff.

File: assets/src/main/java/bisq/asset/coins/PickleProtocol.java (hunk 1 above)
Comment (lilcryptomoon) [reply assets/src/main/java/bisq/asset/coins/PickleProtocol.java]: package bisq.asset.coins;

import bisq.asset.AbstractAssetWithDefaultValidatorTest;

import org.junit.Test;


public class PickleProtocolTest extends AbstractAssetWithDefaultValidatorTest {
    
    public PickleProtocolTest() {
    super(new PickleProtocol());
}
@ Test
    public void testValidAddresses() {
    assertValidAddress("NvkiGCx55W3cGPq3Q2kyQp4dncFypGH7DP");
    assertValidAddress("Nqbi28Hs4WwdqjdEEJWFPg3PryDFReuNgF");
    assertValidAddress("P7nkiTEHb7PWZVxi165tuBwjRhN5XsdYij");
}
@ Test
    public void testInvalidAddresses() {
    assertInvalidAddress("");
    assertInvalidAddress("Pqbi28Hs4WwdqjdEEJWFPg3PryDFReuNgF");
    assertInvalidAddress("N7nkiTEHb7PWZVxi165tuBwjRhN5XsdYij");
}
}

File: assets/src/main/resources/META-INF/services/bisq.asset.Asset (hunk 2 above)
Comment (lilcryptomoon) [reply assets/src/main/resources/META-INF/services/bisq.asset.Asset]: which line Noir ?

File: assets/src/main/java/bisq/asset/coins/PickleProtocol.java (hunk 1 above)
Comment (blabno) [reply assets/src/main/java/bisq/asset/coins/PickleProtocol.java]: I don't know what your coin is, so I am unable to advise you what implementation you should pick.
As to tests you should extend `AbstractAssetTest`.

File: assets/src/test/java/bisq/asset/coins/WrkzCoinTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.AbstractAssetTest;
+
+import org.junit.Test;
+
+public class WrkzCoinTest extends AbstractAssetTest {
```
Comment (blabno) [inline assets/src/test/java/bisq/asset/coins/WrkzCoinTest.java]: Are you sure you want to add this file?

File: assets/src/test/java/bisq/asset/coins/PickleProtocolTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.AbstractAssetWithDefaultValidatorTest;
+
+
+public class PickleProtocolTest extends AbstractAssetWithDefaultValidatorTest {
```
Comment (blabno) [inline assets/src/test/java/bisq/asset/coins/PickleProtocolTest.java:24]: Empty test class?

File: assets/src/main/resources/META-INF/services/bisq.asset.Asset [hunk 5]
```diff
@@ -59,6 +60,7 @@ bisq.asset.coins.TurtleCoin
 bisq.asset.coins.UnitedCommunityCoin
 bisq.asset.coins.Unobtanium
 bisq.asset.coins.Webchain
+bisq.asset.coins.WrkzCoin
```
Comment (blabno) [inline assets/src/main/resources/META-INF/services/bisq.asset.Asset]: Now you're trying to list even more assets.

**Review by blabno — CHANGES_REQUESTED** [review]
NACK per comments

File: assets/src/test/java/bisq/asset/coins/WrkzCoinTest.java (hunk 3 above)
Comment (lilcryptomoon) [reply assets/src/test/java/bisq/asset/coins/WrkzCoinTest.java]: No, the only coin I am trying to add @blabno is my PickleProtcol. Which is Base58 not sure why other people are trying to add coins  to my pull request

**Comment by lilcryptomoon** [comment]
Was trying to pull into this file cleaned up code. but it started a new pull UGH!

**Comment by lilcryptomoon** [comment]
base58 prefix

File: assets/src/test/java/bisq/asset/coins/PickleProtocolTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.junit.Test;
+
+
+public class PickleProtocolTest extends AbstractAssetWithDefaultValidatorTest {
+    
+    public PickleProtocolTest() {
+    super(new PickleProtocol());
+}
+@ Test
```
Comment (blabno) [inline assets/src/test/java/bisq/asset/coins/PickleProtocolTest.java]: Apply Bisq code style

**Review by blabno — CHANGES_REQUESTED** [review]
NACK per comments

File: assets/src/test/java/bisq/asset/coins/PickleProtocolTest.java (hunk 6 above)
Comment (lilcryptomoon) [reply assets/src/test/java/bisq/asset/coins/PickleProtocolTest.java]: Renamed to  List PickleProtocol (PCK)

**Comment by blabno** [comment]
I will vote against this asset as commit history of this PR clearly shows lack of skills.

**Comment by lilcryptomoon** [comment]
Really ? We will close our pull request then @blabno

**Comment by lilcryptomoon** [comment]
@blabno  you may delete the pull request

**Pull request closed without merging**